components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -4178,6 +4178,14 @@ public void checkAppAttributes(Application application, String userId) throws AP
         Map<String, String> applicationAttributes = application.getApplicationAttributes();
         List attributeKeys = new ArrayList<String>();
```

Comment:
how often this method is getting called? Is it when creating a application or retrieving applications or?

Reply:
This method gets called when retrieving applications.